### PR TITLE
Client set name and version on connection

### LINKED ISSUE #1338: Clients should identify themselves on connect
[CLIENT SETINFO](https://redis.io/commands/client-setinfo/) is a new fire-and-forget command that is available as part of redis 7.2. Clients should send the lowercased version of the client name (i.e redis-py, jedis, go-redis) as part of the initial connect, after authenticating to the server.

### PR DESCRIPTION
**Pull request by vladvildanov**
Closes #1338

### CODE REVIEW

**Comment by vladvildanov** [comment]
@tillkruss Adding a fix for Relay connection, leads to an error in CI

https://github.com/predis/predis/actions/runs/5678238422/job/15388057449?pr=1347

Trying to run Relay tests locally, but have a problems with relay installation over homebrew.

`Error: An exception occurred within a child process:
  MachO::DylibUnknownError: No such dylib name: /usr/local/lib/libhiredis_ssl.dylib.1.1.0`
  
 This file actually exists and I have correct version of `hiredis`

**Comment by tillkruss** [comment]
Aah, maybe Relay is already setting `CLIENT SET-NAME` is causing the issue.

>Trying to run Relay tests locally, but have a problems with relay installation over homebrew.
>Error: An exception occurred within a child process: MachO::DylibUnknownError: No such dylib name: /usr/local/lib/libhiredis_ssl.dylib.1.1.0

Relay via Homebrew uses OpenSSL v3, can you try `brew uninstall relay` and reinstall it?

File: src/Connection/Factory.php [hunk 1]
```diff
@@ -174,6 +175,16 @@ protected function prepareConnection(NodeConnectionInterface $connection)
             );
         }
 
+        if (!$connection instanceof RelayConnection) {
```
Comment (chayim) [inline src/Connection/Factory.php:178]: Could these all be pipelined? Speed up the connection a bit?

**Review by chayim — APPROVED** [review]

**Review by tillkruss — APPROVED** [review]

**Comment by BafS** [comment]
1. In this PR add 2 commands on initialization, that's a lot of IO. It should at least be an option IMO. We saw an important load increase in production due to this PR.
2. Why was it bundled in a patch version?

**Comment by tillkruss** [comment]
>We saw an important load increase in production due to this PR.

Can you elaborate on this and provide numbers?

**Comment by BafS** [comment]
We didn't have "client" call before

image

And here you can see the substantial change in CPU in the cluster:

image

it took us a while to find the culprit because we didn't expect to have a change in behavior in a patch version

**Comment by vladvildanov** [comment]
@tillkruss I talked about this with other client-developers and they confirm that this should be an optional. 

We can't fix it other way since it's a Redis API and we must call it twice each time. Let's merge #1370, but before we need to merge #1369 that fixes CI after Redis 7.2 update

**Comment by BafS** [comment]
Thank you for your feedback and thanks for helping fixing it.

Don't you think that this PR should be reverted for the next patch release (to revert the issue introduced) and the new feature (with the new `set_client_info` defaulting to `false` to not have BC) should be in a new minor version?

**Comment by vladvildanov** [comment]
@BafS Totally works for me, but @tillkruss is the releases responsible person

**Pull request merged**